File: examples/hello-world/hello.ts
import { DeltaMemory } from 'deltamemory';

const db = new DeltaMemory({
  apiKey: process.env.DELTAMEMORY_API_KEY,
  baseUrl: process.env.DELTAMEMORY_URL,
  defaultCollection: 'hello-world'
});

async function main() {
  // Store a memory
  console.log('Storing memory...');
  await db.ingest('I love TypeScript and building AI applications');
  
  // Recall the memory
  console.log('Recalling memory...');
  const result = await db.recall('What do I love?');
  
  // Print results
  console.log('\n=== Memories ===');
  result.results.forEach(({ memory, cognitive_score }) => {
    console.log(`[${cognitive_score.toFixed(2)}] ${memory.content}`);
  });
  
  console.log('\n=== User Profile ===');
  result.profiles?.forEach(profile => {
    console.log(`${profile.topic}::${profile.sub_topic}: ${profile.content}`);
  });
}

main().catch(console.error);

File: examples/hello-world/hello.py
import asyncio
import os
from deltamemory import DeltaMemory

async def main():
    db = DeltaMemory(
        api_key=os.environ.get('DELTAMEMORY_API_KEY'),
        base_url=os.environ.get('DELTAMEMORY_URL'),
        default_collection='hello-world'
    )
    
    async with db:
        # Store a memory
        print('Storing memory...')
        await db.ingest('I love Python and building AI applications')
        
        # Recall the memory
        print('Recalling memory...')
        result = await db.recall('What do I love?')
        
        # Print results
        print('\n=== Memories ===')
        for item in result.results:
            print(f"[{item.cognitive_score:.2f}] {item.memory.content}")
        
        print('\n=== User Profile ===')
        if result.profiles:
            for profile in result.profiles:
                print(f"{profile.topic}::{profile.sub_topic}: {profile.content}")

if __name__ == '__main__':
    asyncio.run(main())
